src/js/components/DataTable/DataTable.js

```diff
@@ -422,6 +422,13 @@ const DataTable = ({
     );
   }
 
+  const handleSelect = (nextSelected, row) => {
+    setSelected(nextSelected);
+    if (setSelectedDataContext) setSelectedDataContext(nextSelected.length);
+    if (row) onSelect(nextSelected, row);
```

Comment:
Do we need this added condition or can we stick with old logic of `else onSelect(nextSelected, row)`?

It seems like if `row` is undefined it would be the same either way

Reply:
Fails a Jest test titled "DataTable › onSelect select/unselect all."

Can look at whether the test assertion is correct or not.

Reply:
Confirmed we do need this condition and did exist previously in the body of the table.